secant/Util/UserPreferencesStorage.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  UserPreferencesStorage.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/18/2022.
+//
+
+import Foundation
+
+/// Representation of the user preferences stored in the local persistent storage (non-encrypted, no security needed)
+protocol UserPreferences {
+    /// From when the app is on and uninterrupted
+    var activeAppSessionFrom: TimeInterval { get set }
+    /// What is the set up currency
+    var currency: String { get set }
+    /// Whether the fiat conversion is on/off
+    var isFiatConverted: Bool { get set }
+    /// Whether user finished recovery phrase backup test
+    var isRecoveryPhraseTestCompleted: Bool { get set }
+    /// Whether the user has been autoshielded in the running session
+    var isSessionAutoshielded: Bool { get set }
+}
+
+/// Live implementation of the `UserPreferences` using User Defaults
+/// according to https://developer.apple.com/documentation/foundation/userdefaults
+/// the UserDefaults class is thread-safe.
+struct UserPreferencesStorage: UserPreferences {
+    enum Constants: String, CaseIterable {
+        case zcashActiveAppSessionFrom
+        case zcashCurrency
+        case zcashFiatConverted
+        case zcashRecoveryPhraseTestCompleted
+        case zcashSessionAutoshielded
+    }
+
+    static let `default` = UserPreferencesStorage(
+        appSessionFrom: Date().timeIntervalSince1970,
+        convertedCurrency: "USD",
+        fiatConvertion: true,
+        recoveryPhraseTestCompleted: false,
+        sessionAutoshielded: true,
+        userDefaults: UserDefaults.standard
+    )
+    
+    /// Default values for all preferences in case there is no value stored (counterparts to `Constants`)
+    private let appSessionFrom: TimeInterval
+    private let convertedCurrency: String
+    private let fiatConvertion: Bool
```

Comment:
If we stick with this protocol approach, I don't think you need to hold onto these values.  You could instead just register the defaults values with UserDefaults every time this is initialized, which would also simplify your getters and setters.